### DISCUSSION TITLE
Second order derivatives

### DISCUSSION THREAD

**Discussion by tomix1024** [mitsuba-renderer/mitsuba3#585] [Q&A] [answered]
Hi there,

I was just wondering, if it is possible to compute second order derivatives using drjit?
E.g. if I have two variables `a` and `b` and compute some function `f(a,b)`, I can simply compute `df(a,b)/da` for example using `dr.forward_to`.
But what if I now want to compute the derivative wrt. `b` of the resulting gradient value (`df(a,b)/(da*db)`)?

Here is a simple example of what I want to do:
```python
import mitsuba as mi
import drjit as dr

mi.set_variant('cuda_ad_rgb')

a = mi.Float(3)
b = mi.Float(2)

# compute dab/da
dr.set_grad_enabled(a, True)
dr.set_grad(a, mi.Float(1))
ab = a*b
dr.forward_to(ab)
dab_da = dr.grad(ab)
print(f'{dab_da=}')


# TODO compute [dab/da]/db
dr.set_grad_enabled(b, True)
dr.backward_from(dab_da)
ddab_da_db = dr.grad(b)
print(f'{ddab_da_db=}')
```

Regards,
Tom

**Comment by njroussel** [comment, marked as answer]
Hi @tomix1024 

Drjit doesn't currently support second order derivatives. The few places where we need them we typically the first derivative "manually" and let the autodiff handle the second one.
